# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: browser/base/content/tabbrowser.js
# Commit: 0a3da2090e48
# Full Hash: 0a3da2090e4859473f52b585bcb39baae2876b28
# Author: Jared Wein <jwein@mozilla.com>
# Date: 2021-02-26 03:55:47
# Regressor Bug: 1665390
# File Overlap Count: 1
# Description:
#   Bug 1665390 - Style bookmarks toolbar and tabs toolbar tooltips. r=NeilDeakin,ntim
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D104257
# ==============================================================================

diff -r 176b788753e8 -r 0a3da2090e48 browser/base/content/tabbrowser.js
--- a/browser/base/content/tabbrowser.js	Thu Feb 25 20:47:15 2021 +0000
+++ b/browser/base/content/tabbrowser.js	Thu Feb 25 20:47:54 2021 +0000
@@ -5159,7 +5159,15 @@
         label = this.getTabTooltip(tab);
       }
 
-      event.target.setAttribute("label", label);
+      if (!gProton) {
+        event.target.setAttribute("label", label);
+        return;
+      }
+
+      let title = event.target.querySelector(".places-tooltip-title");
+      title.value = label;
+      let url = event.target.querySelector(".places-tooltip-uri");
+      url.value = tab.linkedBrowser?.currentURI?.spec;
     },
 
     handleEvent(aEvent) {